### PR TITLE
feat(resource-groups): add per-group-type templates

### PR DESCRIPTION
**Pull request by hazedav**
## Summary
This pull-request adds a template for each RGv2 type (i.e. `AWS`, `GCP`, `AZURE`, `OCI`, `MACHINE`, `CONTAINER`).  This makes it easier for our customers to create resource groups via the CLI+Editor

## How did you test this change?
- Manually
- Augmented delete integration test to iterate through each type

## Issue
https://lacework.atlassian.net/browse/GROW-2489

### CODE REVIEW

**Comment by hazedav** [comment]
integration all

**Comment by hazedav** [comment]
integration all

**Comment by hazedav** [comment]
integration all

File: api/resource_groups.go [hunk 1]
```diff
@@ -69,29 +70,57 @@ const (
...
+	GcpResourceGroupQueryTemplate string
+	//go:embed _templates/resource_groups/machine.json
+	MachineResourceGroupQueryTemplate   string
+	LwAccountResourceGroupQueryTemplate string = ""
+	//go:embed _templates/resource_groups/oci.json
+	OciResourceGroupQueryTemplate string
+)
+
+type resourceGroupContext struct {
+	typ           string
```
Comment (zekisherif) [inline api/resource_groups.go:92]: Can we name this something other `typ`? `resourceGroupType`?

File: api/_templates/resource_groups/azure.json [hunk 2]
```diff
@@ -0,0 +1,70 @@
+{
+    "filters": {
+      "filter0": {
+        "field": "Subscription ID",
+        "operation": "EQUALS",
+        "values": [
+          "0fe75302-1906-45ec-bde1-79b76899dd74"
```
[the remark this answers is not in the document]
Comment (hazedav) [reply api/_templates/resource_groups/azure.json:7]: This is all anonymized.

File: api/resource_groups.go (hunk 1 above)
Comment (hazedav) [reply api/resource_groups.go:92]: If we must...

Comment (hazedav) [reply api/resource_groups.go]: Fixed.

**Review by zekisherif — APPROVED** [review]
LGTM

**Review by afiune — APPROVED** [review]

**Pull request merged**